src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -377,15 +377,18 @@ def authorize(self, auth_id, access_level, tenant_id=None, allow_existing_id=Fal
                     tenant=auth_meta['tenant_id']
                 ))
                 auth_meta['dirty'] = True
-                auth_meta['volumes'].update(subvolume)
+                # Update 'volumes' key which was used in ceph_volume_client to 'subvolumes' key
+                if 'volumes' in auth_meta:
+                    auth_meta['subvolumes'] = auth_meta.pop('volumes')
```

Comment:
Order is not retained for old style auth metadata file. Since we always load it into dictionary and modify, I think it should be fine.